### ISSUE TITLE
Grading based on HW2 solutions

### ISSUE THREAD

**Issue by remustawfik** [dilevin/computer-graphics-ray-tracing#37]
Hi,

I just wanted to know if we will be graded with the supplied HW2 solutions build or based on our own HW2 solutions. Thank you.

**Comment by darren-moore** [comment]
You should submit all files, as required on MarkUs, but we will use the supplied HW2 library for grading.

**Comment by radiolariax** [comment]
could someone clarify how the HW2 reference code is accessed via hw2lib? is it just something done at compile time?

**Comment by remustawfik** [comment]
Yes also as a follow up, is it possible to create a build-release with the correct HW2 solutions?
Thank you

**Comment by darren-moore** [comment]
Unfortunately, you can't access the HW2 source code from the `lib`. [Recall how the compilation+linking process works](https://stackoverflow.com/questions/6264249/how-does-the-compilation-linking-process-work).

The reference code is accessed at _link_ time since we're using a static link library. You might contrast this to a dynamic link library (DLL), where the program will only see the code at runtime. "[Static linking vs dynamic linking](https://stackoverflow.com/questions/1993390/static-linking-vs-dynamic-linking)".

**Comment by darren-moore** [comment]
@remustawfik You should be able to make a release build by referencing the release lib.
e.g. `cmake -DCMAKE_BUILD_TYPE=Release -DHW2LIB_DIR=../lib/release/linux/ ..`

Was that your question?
